lib/src/node.dart

```diff
@@ -56,30 +67,100 @@ class StringNode extends Node {
         yamlLines.add(lines[index]);
       }
     } else {
-      var containsSingleQuote = text.contains("'");
-
-      if (context.allowUnquotedStrings &&
-          !containsSingleQuote &&
-          _isValidUnquotedString(text)) {
-        yamlLines.add(text);
-      } else if (!containsSingleQuote) {
-        yamlLines.add('\'$text\'');
+      final containsSingleQuote = text.contains("'");
+      final containsDoubleQuote = text.contains('"');
+
+      if (containsSingleQuote && containsDoubleQuote) {
+        // if contains both single and double quote, escape the double quote,
+        // and use double quote
+        final result = text.replaceAll(r'\', r'\\').replaceAll('"', r'\"');
+        yamlLines.add('"$result"');
```

Comment:
Use `context.config.quoteStyle` to decide which quote to use for this string case.

Reply:
I don't think single quote can be used for this case. Could you give me an example that single quote works in both single and double quote exisiting suitation?

Reply:
Well, this is a single line string, so it won't escape \r, \n, \t...

It only need to escape:

-  double quotes ("): \"
- single quotes ('): '' // double single quote escapes it

So it's possible to maintain the QuoteStyle.

Reply:
Do you think that we need to maintain the `QuoteStyle` in this case?

Reply:
Done